docs/rules/no-container.md

```diff
@@ -0,0 +1,36 @@
+# Disallow the use of `container` methods (no-container)
+
+By using `container` methods like `.querySelector` you may lose a lot of the confidence that the user can really interact with your UI. Also, the test becomes harder to read, and it will break more frequently.
+
+## Rule Details
+
+This rule aims to disallow the use of `container` methods in your tests.
+
+Examples of **incorrect** code for this rule:
+
+```js
+const { container } = render(<Example />);
+const button = container.querySelector('.btn-primary');
+```
+
+```js
+const { container: alias } = render(<Example />);
+const button = alias.querySelector('.btn-primary');
+```
+
+Examples of **correct** code for this rule:
+
+```js
+render(<Example />);
+screen.getByRole('button', { name: /click me/i });
+```
+
+If you use [custom render functions](https://testing-library.com/docs/example-react-redux) then you can set a config option in your `.eslintrc` to look for these.
+
+```
+   "testing-library/no-container": ["error", {"renderFunctions":["renderWithRedux", "renderWithRouter"]}],
```

Comment:
Nice! Thanks for extracting the utils methods for working with this. We may need to set this globally in the plugin through ESLint settings in the future @nickmccurdy @timdeschryver @gndelia